src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -3934,10 +3930,11 @@ private static void ReplaceAdditionalDeclarations(IndentedTextWriter writer, Has
         private static string Literal(RegexOptions options)
         {
             string s = options.ToString();
-            if (int.TryParse(s, out _))
+            if (int.TryParse(s, NumberStyles.Integer, CultureInfo.InvariantCulture, out _))
             {
                 // The options were formatted as an int, which means the runtime couldn't
                 // produce a textual representation.  So just output casting the value as an int.
+                Debug.Fail("This shouldn't happen, as we should only get to the point of emitting code if RegexOptions was valid.");
```

Comment:
Nit: seems like we only have tests that parse invalid int-casting options. Should we also add ctor test where we use valid ones too? ( No need to add those here)

Reply:
We can.  We have several thousand tests that supply valid RegexOptions :smile: but we could certainly add some explicit ones to ensure we successfully accept every valid option.